src/bin/cargo-fmt.rs

```diff
@@ -42,13 +52,26 @@ fn main() {
 
     if verbose && quiet {
         print_usage(&opts, "quiet mode and verbose mode are not compatible");
-        return;
+        return failure;
     }
 
     if matches.opt_present("h") {
         print_usage(&opts, "");
-    } else {
-        format_crate(&opts, verbose, quiet);
+        return success;
+    }
+
+    match format_crate(&opts, verbose, quiet) {
+        Err(e) => {
+            print_usage(&opts, &e.to_string());
+            return failure;
+        }
+        Ok(status) => {
+            if !status.success() {
+                return status.code().unwrap_or(failure);
+            }
```

Comment:
Yeah, mixing `match` and early return on error was a terrible choice :) Fixed and rebased.